gap/examples.gi

```diff
@@ -370,3 +370,33 @@ GeneralisedPetersenGraphCons);
 
 InstallMethod(GeneralisedPetersenGraph, "for integer, integer", [IsInt, IsInt],
 {n, k} -> GeneralisedPetersenGraphCons(IsImmutableDigraph, n, k));
+
+InstallGlobalFunction(DIGRAPHS_KingsGraph,
+function(n, k)
+  local D, i, j;
+  if not IsPosInt(n) or not IsPosInt(k)  then
+    ErrorNoReturn("the arguments <n> and <k> must be positive integers,");
+  fi;
+  D := EmptyDigraph(IsMutable, n * k);
+  for i in [1 .. k] do
+    for j in [1 .. (n - 1)] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + 1]);
+    od;
+  od;
+  for i in [1 .. (k - 1)] do
+    for j in [1 .. n] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + n]);
+    od;
+  od;
+  for i in [1 .. (k - 1)] do
+    for j in [1 .. (n - 1)] do
+      DigraphAddEdge(D, [((i - 1) * n) + j, ((i - 1) * n) + j + (n + 1)]);
+    od;
+  od;
+  for i in [1 .. (k - 1)] do
+    for j in [1 .. (n - 1)] do
+      DigraphAddEdge(D, [((i - 1) * n) + j + 1, ((i - 1) * n) + j + n]);
+    od;
+  od;
+  return DigraphSymmetricClosure(D);
```

Comment:
Again, based on our conversation today, the standard is to return an immutable digraph, and have MutableKingsGraph return a mutable version. (James does this for Pancake as an example).

Reply:
The main function where the work happens should construct a mutable digraph, the default function just calls that function and makes the returned mutable digraph immutable, then returns that. Again I think you need to follow the same pattern as in the `Folkman` PR